### PR TITLE
Focus message input field when starting to type

### LINKED ISSUE #1142: Focus message input field when starting to type
There is no way that I see to set focus to the message input field with the keyboard. One has to use the mouse and click on the field.
I propose to set focus on keypress, so you never have to bother and typing just works.

### PR DESCRIPTION
**Pull request by avoidr**
Closes #1142

---
This change is [Reviewable](https://reviewable.io/reviews/utox/utox/1382)

### CODE REVIEW

**Comment by avoidr** [comment]
Not yet.

**Comment by avoidr** [comment]
I will clean the history in another PR/branch before I merge, but here is my result. I think I'm done now.

File: src/xlib/event.c [hunk 1]
```diff
@@ -415,29 +415,56 @@ bool doevent(XEvent *event) {
...
+                break;
+            }
+
+            if (!edit_active()) {
+                if (messages_char(sym)) {
                     redraw();
                     break;
                 }
+
+                if (ev->state & 4) {
```
Comment (GrayHatter) [inline src/xlib/event.c]: can we make this a macro? I assume that `& 4` tests for ctrl?

File: src/xlib/event.c [hunk 2]
```diff
@@ -415,29 +415,56 @@ bool doevent(XEvent *event) {
...
+                /* Focus message input field if ctrl isn't pressed,
+                 * to make sure you can still copy text from the chat log */
+                if (sym != XK_Control_L) {
+                    edit_setfocus(&edit_chat_msg_friend);
+                    edit_char(KEY_END, 1, 0);
+                }
             }
 
             if (edit_active()) {
+                if (sym == XK_Escape) {
```
Comment (GrayHatter) [inline src/xlib/event.c]: what's Esc doing here?

File: src/xlib/event.c (hunk 1 above)
Comment (avoidr) [reply src/xlib/event.c]: Yeah, actually that is `ControlMask` from Xlib.

File: src/xlib/event.c (hunk 2 above)
Comment (avoidr) [reply src/xlib/event.c]: It's explained in the commit message, that block is not directly related to focusing, somewhat a separate "feature".

**Comment by avoidr** [comment]
Cleaned up and merged.

**Pull request closed without merging**